### ISSUE TITLE
reproducibility issue: unable to reproduce DSC and HD for testing dataset

### ISSUE THREAD

**Issue by ghafoorharis** [danivelikova/lotus#4]
I have obtained inference for  `GT_data_vessels_testing_100` as recommended. The function i used to calcualte the metrics is attached below(maybe i will raise PR). Could you please identify what went wrong. Please note that I retrained LOTUS from end to end using your scripts. 

`
def calculate_metrics_ind(outputs, targets, threshold=0.5):
    """
    Calculates evaluation metrics for binary segmentation per image and averages over the batch.
    Args:
        outputs (torch.Tensor): Model predictions, expected to be probabilities in [0,1], shape [batch_size, 1, H, W]
        targets (torch.Tensor): Ground truth labels, shape [batch_size, 1, H, W]
    Returns:
        metrics_dict (dict): Dictionary containing averaged 'dice', 'iou', 'precision', 'recall', 'accuracy'
    """
    epsilon = 1e-7  # Small constant to avoid division by zero
    # Threshold outputs to obtain binary predictions
    # Downsample the outputs and targets to 256x256
    outputs_bin = (outputs > threshold).float()
    targets_bin = (targets > threshold).float()

    batch_size = outputs.size(0)
    dice_scores = []
    iou_scores = []
    precision_scores = []
    recall_scores = []
    accuracy_scores = []
    hd95_scores = []
    # new two metrics for roc curve
    fpr = []
    tpr = []
    # make the loop to be tqdm
    for i in tqdm(range(batch_size)):
        output_i = outputs_bin[i].view(-1)
        target_i = targets_bin[i].view(-1)

        # True Positives, False Positives, True Negatives, False Negatives
        TP = (output_i * target_i).sum()
        FP = (output_i * (1 - target_i)).sum()
        TN = ((1 - output_i) * (1 - target_i)).sum()
        FN = ((1 - output_i) * target_i).sum()

        # Dice Coefficient
        dice = (2 * TP + epsilon) / (2 * TP + FP + FN + epsilon)
        dice_scores.append(dice.item())

        # Intersection over Union (IoU)
        iou = (TP + epsilon) / (TP + FP + FN + epsilon)
        iou_scores.append(iou.item())

        # Precision
        precision = (TP + epsilon) / (TP + FP + epsilon)
        precision_scores.append(precision.item())

        # Recall
        recall = (TP + epsilon) / (TP + FN + epsilon)
        recall_scores.append(recall.item())

        # Accuracy
        accuracy = (TP + TN + epsilon) / (TP + TN + FP + FN + epsilon)
        accuracy_scores.append(accuracy.item())

        # Compute HD95
        hd95 = compute_hd95(outputs_bin[i].squeeze(0), targets_bin[i].squeeze(0))
        # hd95 = 0
        hd95_scores.append(hd95)

        # Compute FPR and TPR
        # Detach the FP, TN, TP, FN
        FP = FP.detach().cpu().numpy()
        TN = TN.detach().cpu().numpy()
        TP = TP.detach().cpu().numpy()
        FN = FN.detach().cpu().numpy()
        fpr.append(FP / (FP + TN))
        tpr.append(TP / (TP + FN))
        # print the fpr and tpr
        # print(f"FPR: {fpr[-1]}, TPR: {tpr[-1]}")

    # Average over batch
    metrics_dict = {
        "dice": dice_scores,
        "iou": iou_scores,
        "precision": precision_scores,
        "recall": recall_scores,
        "accuracy": accuracy_scores,
        "hd95": hd95_scores,
        "fpr": fpr,
        "tpr": tpr,
    }

    return metrics_dict
`

Results: 

<img width="812" height="322" alt="Image" src="https://github.com/user-attachments/assets/0b7ab99e-2414-498a-b34a-350b2fefc395" />

The results are significantly different than expected. Your feedback will be highly appreciated. 

Thanks
